src/sentry/search/events/fields.py

```diff
@@ -1314,6 +1314,24 @@ def with_default(default, argument):
     return argument
 
 
+class SnQLNumericColumnNoLookup(NumericColumn):
```

Comment:
Could we just use Numeric Column here instead of introducing a new Column type? if this is for the type check whether we resolve args, I think we'll want to resolve anything that's a column not just the ones we've called out as `SnQL...Column`s

Reply:
I don't think we can just use NumericColumn here since because of this return [statement](https://github.com/getsentry/sentry/blob/master/src/sentry/search/events/fields.py#L1176). We'd just want to return the original value (which is the alias) here and not the mapped Discover column. The `self.column()` call in the QueryFields class will handle that

Reply:
Hmm I still think we don't need to introduce new Columns here but let me try that in a separate PR so we can get this change in :+1: